### PR TITLE
`Assessment`: Fix course scores export if user information is not set

### LINKED ISSUE #4815: Score Export Broken
### Describe the bug

The points for individual programming exercises in the CSV file exported via `Scores` do not match the displayed points in the UI. For example, we had a student with 100% in the web UI, but according to the CSV, he/she didn't even participate in the exercise. In addition, the sum of exported points does not match `Programming Points` in the CSV. However, `Programming Points` is equal to the web UI.

We assume, that programming points are correctly calculated, but the export of points for individual programming exercises is broken.

Incorrect output:
| Name | Username | ... | Binary Search Tree | Bash | ASCII Tabelle | ... | Programming Points | ... |
| --- | --- | --- | --- | --- | --- | --- | --- | --- |
| xyz | xyz | ... | `<wrong points>` | `<wrong points>` | `<wrong points>` | ... | `<sum>` | ... |

### To Reproduce

1. Go to `Course Management`
2. Select a course with several programming exercises
3. Click on `Scores`
4. Click on `Export as CSV`
5.1 Check whether the sum of points received in individual programming exercises match `Programming Points` 
5.2 Make sure that points attributed to an exercise belong to this exercise.

### Expected behavior

1. Sum of individual programming exercises matches `Programming points`.
2. Points for individual programming exercises belong to the exercise.

### Screenshots

_No response_

### What browsers are you seeing the problem on?

Firefox

### Additional context

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by b-fein**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).

### Motivation and Context
https://github.com/ls1intum/Artemis/issues/4815#issuecomment-1078875569
Closes #4815

### Description
Allow for user information that is not required to be empty. Allows the login name to be empty for consistency as well, even if that shouldn’t really happen.

### Steps for Testing
Prerequisites:
- 1 Instructor
- 1 Course with student participations

1. Edit the user information of a student so that the name or email is missing.
2. Go to Course -> Scores and click on Export Results. The results should be exported with empty fields in the CSV where the user information should be. No error should be shown.

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
Optional, fix should be clear from a code review.
- [ ] Test 1
- [ ] Test 2

### Test Coverage
All new code is tested by adding a unit test specifically for this case.

### Screenshots
n/a

### CODE REVIEW

File: src/main/webapp/app/course/course-scores/course-scores-csv-row-builder.ts [hunk 1]
```diff
@@ -41,7 +41,11 @@ export class CourseScoresCsvRowBuilder {
      * @param value That should be placed in the row. Replaced by the empty string if undefined.
      */
     set<T>(key: string, value: T) {
-        this.csvRow[key] = value ?? '';
```
Comment (Strohgelaender) [inline src/main/webapp/app/course/course-scores/course-scores-csv-row-builder.ts:44]: Why is the ??-operator not possible here?

Comment (b-fein) [reply src/main/webapp/app/course/course-scores/course-scores-csv-row-builder.ts]: I did not intend that change and it is a left-over from something else I was testing. Reverted in the latest commit.

**Review by Strohgelaender — APPROVED** [review]
Code looks good and the test passes -> approve

**Review by bassner — APPROVED** [review]
Code looks good!

**Pull request merged**